src/test/java/net/snowflake/client/jdbc/DatabaseMetaDataLatestIT.java

```diff
@@ -1694,4 +1694,172 @@ public void testUpdateLocatorsCopyUnsupported() throws SQLException {
       assertFalse(metaData.locatorsUpdateCopy());
     }
   }
+
+  @Test
+  public void testNoPatternSearchAllowedForPrimaryAndForeignKeys() throws SQLException {
+    Map<String, String> params = getConnectionParameters();
+    Properties properties = new Properties();
+    for (Map.Entry<?, ?> entry : params.entrySet()) {
+      if (entry.getValue() != null) {
+        properties.put(entry.getKey(), entry.getValue());
+      }
+    }
+    // test out connection parameter stringsQuoted to remove strings from quotes
+    properties.put("enablePatternSearch", "false");
+    Connection connection = DriverManager.getConnection(params.get("uri"), properties);
+
+    String database = connection.getCatalog();
+    String schema = connection.getSchema();
```

Comment:
the problem we are trying to solve is related also to the cases where schema and/or table name contains wildcards (`_` or `%`) - please also add such test cases (remember to create new schema and use it in closure via `TestUtil.withSchema` with test schema generated with prefix as in another tests)